lib/Migrations/Version20170927125822.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace OCA\User_LDAP\Migrations;
+
+use OCA\User_LDAP\Helper;
+use OCP\Migration\ISimpleMigration;
+use OCP\Migration\IOutput;
+use \OCA\User_LDAP\Configuration;
+
+/**
+ * Migration to move the ldap config IDs to all use a prefix for consistency
+ */
+class Version20170927125822 implements ISimpleMigration {
+
+	/**
+	 * @var Helper
+	 */
+	protected $helper;
+
+	public function __construct(Helper $helper) {
+		$this->helper = $helper;
+	}
+
+	/**
+     * @param IOutput $out
+     */
+    public function run(IOutput $out) {
+        // Find out what configurations we are using and modify them
+
+		$configPrefixes = $this->helper->getServerConfigurationPrefixes();
+		foreach($configPrefixes as $prefix) {
+			if($prefix === '') {
+				// Found an empty prefix - we need to modify this one
+				$newPrefix = $this->helper->getNewConfigPrefix();
+				$oldConfig = new Configuration('');
+				// Migrate the config
+				$newConfig = new Configuration($newPrefix, false);
+				foreach($oldConfig->getDefaults() as $configKey) {
+					// Set the new config equal to the old config, or default
+					$newConfig->$configKey = $oldConfig->$configKey;
+				}
+				$newConfig->saveConfiguration();
+				$this->helper->deleteServerConfiguration('');
+			}
+		}
```

Comment:
Before the migration - the app is loaded by the MigrationService, which will initialise the User backend using the old prefix. So after the migration we need to before this configuration from the user proxy and insert a new one - otherwise things will start to 💥

Reply:
Could add a method to User_Proxy to remove a backend given a configuration...

Reply:
Cache will be invalidated because the configuration prefix is used in the cache key https://github.com/owncloud/user_ldap/blob/9e3117d6a03136d14cc4aa409d7bed4f342feb21/lib/Connection.php#L200 - could lead to longer migrations / upgrades after this migration is executed

Reply:
If we tie this PR to #125 then caching is not an issue - unless many groups are present

Reply:
Caching shouldn't be that much of an issue for appliance use case - they will most likely not have many users/groups. But we need to make sure that #125 and this work together when #125 gets merged